doc/effects.md

```diff
@@ -180,14 +180,67 @@ Usage example:
 ```dart
 final combination = CombinedEffect(
     effects: [move, scale, rotate],
-    isInfinite: true, 
+    isInfinite: true,
     isAlternating: true);
 myComponent.addEffect(combination);
 ```
 
 An example of how to use the `CombinedEffect` can be found
 [here](https://github.com/flame-engine/flame/tree/main/examples/lib/stories/effects/combined_effect.dart).
 
+## Common for paint effects
+
+Flame provides an useful mixin that adds paint variables to your components, that is used by some
+of Flame's own components like `SpriteComponent`, `SpriteAnimationComponent`, but you can also use
+it on you any custom component that your game has, and any component that uses that mixin can have
+paint affects applied to it.
+
+By default all effects will animate the main paint of the component, that can be changed by passing
+a `paintId` to the effect contructor. This can be useful for when you have a component that have
+multiple paints, for example a component which has a foreground and background layer.
```

Comment:
```suggestion
By default, all effects will animate the main paint of the component. That can be changed by passing
a `paintId` [note: or key?] to the effect contructor. This can be useful for when you have a component that have
multiple paints; for example, a component which has a foreground and background layer.
```

I think before this paragraph explaining how effects behaves wrt to multiple paints, we should have a paragraph dedicated to explaining how the `HasPaint` mixing allows for one main paint and other, optional sub-paints that you can key by providing a generic type parameter (because that is not super easy to grasp so it's good to be precise before even talking about effects).